hazelcast-spring/src/main/java/com/hazelcast/spring/cache/HazelcastCacheManager.java

```diff
@@ -70,11 +83,59 @@ public Cache getCache(String name) {
         return cacheNames;
     }
 
-    public void setHazelcastInstance(final HazelcastInstance hazelcastInstance) {
-        this.hazelcastInstance = hazelcastInstance;
+    private long calculateCacheReadTimeout(String name) {
+        long timeout = 0;
+        if (getReadTimeoutMap().containsKey(name)) {
+            Long v = getReadTimeoutMap().get(name);
+            if (v != null) {
+                timeout = v;
+            }
+        } else {
+            timeout = defaultReadTimeout;
+        }
+        return timeout;
```

Comment:
setter or before the setter... somehow while parsing XML. However, if you consider `null` values are valid values and interpret as 0 then I am fine with it. User should know about leaving it blank will make timeout `0` in the documentation.